automation_tools/__init__.py

```diff
@@ -573,31 +573,69 @@ def install_puppet_scap_client():
     run('yum -y install puppet-foreman_scap_client', warn_only=True)
 
 
-def setup_foreman_discovery():
-    """Task to setup foreman discovery."""
-    # Install required packages to enable discovery plugin
+def setup_foreman_discovery(sat_version, upstream=False):
+    """Task to setup foreman discovery.
+
+    :param str sat_version: should contain satellite version e.g. 6.1, 6.2
+    :param bool upstream: Flag whether we run on upstream. Default: ``False``.
+    """
     packages = (
         'tfm-rubygem-foreman_discovery',
         'rubygem-smart_proxy_discovery',
-        'foreman-discovery-image',
         'tfm-rubygem-hammer_cli_foreman_discovery',
     )
-    run('yum install -y {0}'.format(' '.join(packages)))
+    admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
 
-    for daemon in ('foreman', 'httpd', 'foreman-proxy'):
-        manage_daemon('restart', daemon)
+    if not upstream and sat_version in ('6.1', '6.2'):
+        run('yum install -y {0}'.format(' '.join(packages)), warn_only=True)
+        run('yum install -y foreman-discovery-image')
+        for daemon in ('foreman', 'httpd', 'foreman-proxy'):
+            manage_daemon('restart', daemon)
+        template_url = os.environ.get('PXE_DEFAULT_TEMPLATE_URL')
+        template_file = run('mktemp')
+        hostname = run('hostname -f', quiet=True).strip()
+        run('wget -O {0} {1}'.format(template_file, template_url))
+        run('sed -i -e "s/SatelliteCapsule_HOST/{0}/" {1}'
+            .format(hostname, template_file))
+        # Update the template
+        run('hammer -u admin -p {0} template update --name '
+            '"PXELinux global default" --file {1}'
+            .format(admin_password, template_file))
+        run('rm -rf {0}'.format(template_file))
+        return
 
-    admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
-    template_url = os.environ.get('PXE_DEFAULT_TEMPLATE_URL')
-    if template_url is None:
-        print('You must specify the PXE default template URL')
-        sys.exit(1)
+    if upstream:
+        run('yum install -y {0}'.format(' '.join(packages)), warn_only=True)
+        # Fetch the nightly FDI from upstream
+        image_url = 'http://downloads.theforeman.org/discovery/nightly/fdi-image-latest.tar'  # noqa
+        run('wget {0} + -O - | tar x --overwrite -C /var/lib/tftpboot/boot'
+            .format(image_url))
+    if not upstream and sat_version == '6.3':
+        # In 6.3, installer should install all required packages except FDI
+        run('rpm -q {0}'.format(' '.join(packages)))
+        run('yum install -y foreman-discovery-image')
 
+    for daemon in ('foreman', 'httpd', 'foreman-proxy'):
```

Comment:
lines 618 - 640 seems common for 6.3 and for upstream but not for 6.1 and 6.2 
so you wanted to indent these line not be executed twice for 6.1 and 6.2

so you have to join two main ifs into one:
```
if upstream or sat_version == '6.3':
```
and add smaller nested ifs to exlude only for upstream/6.3